ProjectSidewalk/SidewalkWebpage
# 1207 full priority based routing

#### What this PR does
* Updates the front-end routing to be entirely based on priority
* Rewrites the task selection code in AuditTaskTable.scala to assign tasks based on priority instead of completion count. This is the most annoying part to test, because there are something like 6 different cases to test for.
* Removes a few large-ish chunks of code that we are no longer using, and I don't see a reason to keep - let me know if you disagree with any of those choices :)
* Generally cleaned up any code I saw that needed some cleaning... So I removed a bunch of unused import statements (which is why the PR has 11 modified files), fixed misspellings, renamed variable, made some minor changes to syntax to make things more readable/consistent, etc. I'm going to go through the PR and add comments next to anything that is essentially just cosmetic, and I don't think requires testing.

#### Description of how new priority-based routing works
This PR primarily fleshes out the priority based routing that @maddalihanumateja had started working on for the front-end. Users are now always routed to a street that has roughly the same priority level as the highest priority street in the region in which they are auditing. To do this, I discretized the priority (which goes from 0 to 1) into 4 levels; [0,0.25), [0.25,0.5), [0.5,0.75), [0.75,1].

Since our priority system is currently just based on audit count, where priority = 1/(1+audit_count), for audit counts 0, 1, 2, 3, 4, 5, ...; we have priority 1, 0.5, 0.33, 0.25, 0.2, 0.17, ...; this maps into priority levels 3, 2, 1, 1, 0, 0, ... Thus, within a given region, all streets with audit count = 0 will be audited, followed by those with audit count = 1, then those with audit count _either_ 2 _or_ 3, followed by anything else.

The way this works in practice is analogous to how we had talked about routing based on audit count beforehand, @manaswis. Example: If the entire region has been audited completely twice over, then when a user finishes a street, they will only jump if _none_ of the adjacent streets have an audit count of 2 or 3 (i.e., a street with region-wide maximum discretized priority). If the user does not have to jump, we take the highest priority adjacent street. If they do jump, we take the highest priority street in the entire region.

#### How to test
1. For the changes to the functions in AuditTaskTable.scala: I would go to each function that has been rewritten, add a print statement of some sort (to verify that this code was actually run when you try to test it), add in whatever print statements or whatever else you need to assure yourself that it is correct. Each one is sort of different, and I think you should just do whatever you need to feel good about its correctness.

    I made the comment above each of these functions more descriptive, so it now tells you when the code should be called, hopefully making testing easier. For example, I changed one description from "Get a task that is in a given region." to "Get a task that is in a given region. Used if a user has already been assigned a region, or from /audit/region."

2. For the priority-based routing code: I added some console logging statements to help guide you with this. For example, when a user finishes a street, I log the highest priority task in the region, the list of adjacent streets, then the list of adjacent streets after filtering out the ones with priority that is too low. You should go through and make sure the behavior seems correct, in the you get routed down streets with appropriate priority, you are only forced to jump when the priority of adjacent streets is too low, when you do jump you should be sent to a high-priority street, etc.

    In the console, I found the following functions of the Tasks to be useful: isCompleted(), getStreetEdgeId(), getStreetPriority(), getStreetPriorityDiscretized(), getStreetCompletionCount().


Let me know if I can clarify anything. Thanks!

Resolves #1207
